specs/ExtendedProcessFailed.md

```diff
@@ -0,0 +1,671 @@
+            wil::com_ptr<ICoreWebView2ProcessFailedEventArgs> args = argsRaw;
+            COREWEBVIEW2_PROCESS_FAILED_KIND failureKind;
+            CHECK_FAILURE(args->get_ProcessFailedKind(&failureKind));
+            if (failureKind == COREWEBVIEW2_PROCESS_FAILED_KIND_BROWSER_PROCESS_EXITED)
+            {
+                int button = MessageBox(
+                    m_appWindow->GetMainWindow(),
+                    L"Browser process exited unexpectedly.  Recreate webview?",
+                    L"Browser process exited", MB_YESNO);
+                if (button == IDYES)
+                {
+                    m_appWindow->ReinitializeWebView();
+                }
+            }
+            else if (failureKind == COREWEBVIEW2_PROCESS_FAILED_KIND_RENDER_PROCESS_UNRESPONSIVE)
+            {
+                int button = MessageBox(
+                    m_appWindow->GetMainWindow(),
+                    L"Browser render process has stopped responding.  Recreate webview?",
+                    L"Web page unresponsive", MB_YESNO);
+                if (button == IDYES)
+                {
+                    m_appWindow->ReinitializeWebView();
+                }
+            }
+            else if (failureKind == COREWEBVIEW2_PROCESS_FAILED_KIND_RENDER_PROCESS_EXITED)
+            {
+                int button = MessageBox(
+                    m_appWindow->GetMainWindow(),
+                    L"Browser render process exited unexpectedly. Reload page?",
+                    L"Web page unresponsive", MB_YESNO);
+                if (button == IDYES)
+                {
+                    CHECK_FAILURE(m_webView->Reload());
+                }
+            }
+
+            // Check the runtime event args implements the newer interface.
+            auto args2 =
+                args.try_query<ICoreWebView2ProcessFailedEventArgs2>();
+            if (!args2)
+            {
+                return S_OK;
+            }
+
+            if (failureKind ==
+                COREWEBVIEW2_PROCESS_FAILED_KIND_FRAME_RENDER_PROCESS_EXITED)
+            {
+                // A frame-only renderer has exited unexpectedly. Check if reload is needed.
+                wil::com_ptr<ICoreWebView2FrameInfoCollection> impactedFrames;
+                wil::com_ptr<ICoreWebView2FrameInfoCollectionIterator> iterator;
+                CHECK_FAILURE(args2->get_ImpactedFramesInfo(&impactedFrames));
+                CHECK_FAILURE(impactedFrames->GetIterator(&iterator));
+
+                BOOL hasCurrent = FALSE;
+                while (SUCCEEDED(iterator->HasCurrentFrameInfo(&hasCurrent)) && hasCurrent)
+                {
+                    wil::com_ptr<ICoreWebView2FrameInfo> frameInfo;
+                    CHECK_FAILURE(iterator->GetCurrentFrameInfo(&frameInfo));
+
+                    wil::unique_cotaskmem_string nameRaw;
+                    wil::unique_cotaskmem_string sourceRaw;
+                    CHECK_FAILURE(frameInfo->get_Name(&nameRaw));
```

Comment:
We lucked out that we never use `nameRaw`, which might be `null` if the frame had no name.

Should it return an empty string `L""` rather than a null pointer? See discussion at `ProcessDescription` property below.

Reply:
It will return empty string. Updated the docs.